api.go

```diff
@@ -33,37 +33,20 @@ func (a *API) HandleDirectory(dir string) error {
 		return fmt.Errorf("Cannot list clang-c directory: %v", err)
 	}
 
-	h := &HeaderFile{
-		api: a,
-
-		dir: dir,
-
-		lookupEnum:        map[string]*Enum{},
-		lookupNonTypedefs: map[string]string{},
-		lookupStruct: map[string]*Struct{
-			"cxstring": &Struct{
-				Name:  "cxstring",
-				CName: "CXString",
-			},
-		},
-	}
-
 	for _, hf := range headers {
-		name := hf.Name()
-
-		if hf.IsDir() || !strings.HasSuffix(name, ".h") {
+		if hf.IsDir() || !strings.HasSuffix(hf.Name(), ".h") {
 			continue
 		}
-
-		h.name = dir + name
+        
+        h := newHeaderFile(a, dir + hf.Name(), dir)
```

Comment:
I guess I reused the "Clang" API because of "sourceFile.Name()". I think that it would make more sense to refactor this into the HeaderFile members "filename" and "path", and then concatenate them. E.g. `ioutil.ReadFile(h.Name)` would change to `ioutil.ReadFile(h.Path+ "/" + h.Filename)`. Or even add a helper function "FilePath" to HeaderFile, `ioutil.ReadFile(h.FilePath())`. This would make the API more clear. What do you think?